app/models/miq_user_role.rb

```diff
@@ -37,8 +37,8 @@ def feature_identifiers
   def allows?(identifier:)
     if feature_identifiers.include?(identifier)
       true
-    elsif parent = MiqProductFeature.parent_for_feature(identifier)
-      allows?(:identifier => parent.identifier)
+    elsif (parent = MiqProductFeature.feature_parent(identifier))
+      allows?(:identifier => parent)
```

Comment:
I think I see the confusion...

The features method is keyed by and contains values for "identifiers", not actual feature rows:

```ruby
  def self.features
    @feature_cache ||= begin
      includes(:parent, :children).each_with_object({}) do |f, h|
        child_idents = f.children.collect(&:identifier)
        parent_ident = f.parent.identifier if f.parent
        details      = DETAIL_ATTRS.each_with_object({}) { |a, dh| dh[a] = f.send(a) }
        h[f.identifier] = {:parent => parent_ident, :children => child_idents, :details => details}
      end
    end
  end
```

Therefore, we're changing from a method that returns a feature row to a method that returns an identifier:

```ruby
  def self.feature_parent(identifier)
    features[identifier.to_s].try(:[], :parent)
  end

  def self.parent_for_feature(identifier)
    find_by_identifier(feature_parent(identifier))
  end
```